WalletWasabi/Wallets/TransactionFeeProvider.cs

```diff
@@ -101,19 +102,16 @@ protected override async Task ExecuteAsync(CancellationToken cancel)
 		{
 			await Semaphore.WaitAsync(cancel).ConfigureAwait(false);
 
-			if (!Queue.TryDequeue(out var txidToFetch))
+			if (TransactionIdChannel.Reader.TryRead(out var txidToFetch) && txidToFetch is not null)
```

Comment:
Very good. Now you can add Reader.ReadAsync() https://learn.microsoft.com/en-us/dotnet/api/system.threading.channels.channelreader-1.readasync?view=net-8.0

Reply:
And remove the Semaphore.

Reply:
It cannot be null.

Reply:
Yep, after lunch I will implement the local tasks list to limit the parallel requests to 3.

Reply:
https://github.com/molnard/WalletWasabi/pull/61/commits/91a9abf2838f3e2069a6458a933451f6e9f7dfbb